### ISSUE TITLE
[Runtime Bug]: Max Payne 2 No Longer Hooks

### ISSUE THREAD

**Issue by skurtyyskirts** [NVIDIAGameWorks/rtx-remix#552]
### Describe the bug

It looks like Max Payne 2 recently stopped hooking with RTX-Remix. 

### How do you reproduce the bug?

Install dxwrapper with d3d8 enabled, configure like so, and launch game. Game launches, but remix doesn't hook.
![Screenshot 2024-06-30 210024](https://github.com/NVIDIAGameWorks/rtx-remix/assets/148610406/a64443b1-24e8-44ff-b958-1409131f8c39)
![Screenshot 2024-06-30 210029](https://github.com/NVIDIAGameWorks/rtx-remix/assets/148610406/bc871cf9-b994-4594-bc96-7655fb18ee09)

### What is the expected behavior?

It looks like Darko9.13 had it running in Discord just a few months ago, but it recently stopped hooking. They state that it hooks with Portal RTX, but I haven't had such luck. I tried deleting the d3d9to8 that now comes with Remix, but that also didn't work. 

### Version

d42dbe6

### Logs

[D3D9.log](https://github.com/user-attachments/files/16047197/D3D9.log)
[NvRemixBridge.log](https://github.com/user-attachments/files/16047199/NvRemixBridge.log)
[dxwrapper-maxpayne2.log](https://github.com/user-attachments/files/16047200/dxwrapper-maxpayne2.log)

**Comment by NV-LL** [comment]
REMIX-3324 for tracking

**Comment by PappaSlask** [comment]
@skurtyyskirts  Did you rename d3d8tod3d9.dll to just d3d8.dll? Cuz wrapper is disabled by default these days.

**Comment by skurtyyskirts** [comment]
Yes, renamed it to d3d8.dll, also tried other wrappers but no dice for me. Very strange, never had this issue before. Been at it for awhile, not sure what it could be.

**Comment by PappaSlask** [comment]
> Yes, renamed it to d3d8.dll, also tried other wrappers but no dice for me. Very strange, never had this issue before. Been at it for awhile, not sure what it could be.

Could be because of the recent DXVK and Bridge updates... There was a lot of issues for MP1 with those, so I reverted back to older ones. If you want, you could test the ones I'm currently using, I'll attach them to this post.
[DXVK_Bridge_latest_working_versions.7z.001.zip](https://github.com/user-attachments/files/16117249/DXVK_Bridge_latest_working_versions.7z.001.zip)
[DXVK_Bridge_latest_working_versions.7z.002.zip](https://github.com/user-attachments/files/16117250/DXVK_Bridge_latest_working_versions.7z.002.zip)

Just remove the ".zip" part of the filenames to extract. Also, make sure to delete any existing files related to remix, dxvk and bridge from your MP2 folder. Good luck bud!

**Comment by NV-DM** [comment]
@PappaSlask I'm unable to extract the provide files. Do you know if there's a commit that broke it?

**Comment by skurtyyskirts** [comment]
Hi PapaSlask! Thanks for sending those over. I too unfortunately am having issues extracting them. I made sure to remove the .zip extension (making sure I had extensions enabled) and right clicked, and used winrar to open the achieve, but received this error message. 
![Screenshot 2024-07-19 133757](https://github.com/user-attachments/assets/a103d047-aaca-4f4b-9e3e-3c669af48ae3)
![Screenshot 2024-07-19 133809](https://github.com/user-attachments/assets/37158f68-fd1c-4dd3-ac99-09a8c3f2def0)


On the Windows Defender possibility, I went ahead and looked under Protection History and looked to see if maybe it was quarantined or removed, but it looks like no recent history was listed. I performed a regular boot with remix installed, so no Windows Defender protection history, (reinstalled remix just incase) and performed a boot of MP2 with Remix installed with Defender off, and still no hook unfortunately.

Here's my folder structure of MP2, along with my dxwraper.ini
![Screenshot 2024-07-19 133246](https://github.com/user-attachments/assets/14143bf4-57e6-4de7-add2-408bcd10d9f4)
![Screenshot 2024-07-19 133257](https://github.com/user-attachments/assets/4436915e-4810-485c-bdc9-adecd21b0b1d)
![Screenshot 2024-07-19 134232](https://github.com/user-attachments/assets/e0696956-e05d-4210-a993-b56ac287e501)
![Screenshot 2024-07-19 134312](https://github.com/user-attachments/assets/501565ad-c106-4a66-ba76-b99d68d50c95)

**Comment by PappaSlask** [comment]
@skurtyyskirts Hi Skurty!

Sorry for the inconvenience and thanks for getting back to me regarding this! Also thanks for the update regarding the extraction issue. I think I have nailed down what the problem was all along: **You need to have 7zip installed in order to extract these archive files**, since they were created with that program to begin with. So kindly install 7zip (it's tons better than winrar anyway lol) and try the extraction process once again. Remember that you need to right click on the 001 file and go to the 7zip context menu to choose *Unpack to* for this to work.

Thanks for your patience! Also, your MP2 file structure and files look all fine to me. But I want to rise a concern regarding the Steam versions of MP1 and MP2 and the **latest** (and possible even a few updates prior to latest) dxvk/bridge versions! A recent member on our team had issues getting MP1 working with the steam version, and he first used the Remix auto-downloader. But after helping him troubleshooting, I decided to let him try my older version, so I sent them to him and ***kabom*** it worked just fine.

So my early conclusion is that there's a compatibility issue with the recent dxvk/bridge versions AND steam version of these games, though it could be possible that the same is true for ALL copies of these games and not exclusive to the Steam version. I haven't tried the updated dxvk/bridge files myself (I'm using retail-cd version) So I can't confirm that.

**Comment by skurtyyskirts** [comment]
Thanks PappaFlask! I was able to get that remix version you sent over unachieved, but still no dice. I've tried using dxwrapper, crosire's d3d8to9, but still no hook. I've disabled nvidia's display overlay, disabled steam overlay, unplugged my second monitor, downloaded the dx9 runtime files on the MP1 demo, disabling full-screen optimizations, started with shortcut launch options of -skipstartup -window -developer -developerkeys, checked my windows protection history if there were any quarantines but the history was clean. Made an exception rule to exclude MP2 directory, but still no hook. I'm completely stumped, I've never seen this before, been using Remix for almost a year now and have never seen anything like this. 

Maybe it has something to do with the steam version?

**Comment by skurtyyskirts** [comment]
Did a clean install of Max Payne 2, ditched the dxwrapper and used crosire/d3d8to9 wrapper using Remix 0.4 and 0.4.1 but still no hook. I'm not sure if this is a regression issue, I'm confused why the hook isn't happening and if this is maybe something to do with steam or not. 
[D3D9.log](https://github.com/user-attachments/files/16353738/D3D9.log)

**Comment by skurtyyskirts** [comment]
Tried Remix 0.3 on the Max Payne 2 Demo, no luck either. 

[D3D9.log](https://github.com/user-attachments/files/16354362/D3D9.log)

**Comment by PappaSlask** [comment]
@skurtyyskirts , @NV-DM , @NV-LL 

I've found the problem, and a solution! After digging deeper into this issue, I found that it boils down to the d3d8.dll and the Dx8toDx9 conversion. Since about the release of Remix Runtime 0.5.0, a d3d8to9 wrapper is included in the installation files. This particular wrapper is Crosire's own wrapper (the guy behind the SweetFX and Reshade software) and is named d3d8.dll and is about 121kb in size.

This file, this wrapper mentioned above isn't needed at all for any of the Max Payne games! Instead, we should use the wrapper that comes with the ***Definitive Fixes*** [patch.](https://community.pcgamingwiki.com/files/file/2325-max-payne-anthology-definitive-fixes/) Among its files, there's a d3d8.dll which weighs in at roughly 2.1Mb. We should take this file, rename it into dinput.dll, then do either of these 3 following things:
- Make a copy of the dinput.dll and rename it into d3d8.dll
- Rename Remix's d3d8to9.dll to d3d8.dll
- Use Crosire's wrapper (already named d3d8.dll)

Just to avoid any confusion, these .dll files should be in the root game dir, where there game executable is.
***#Final notes:*** You might, or might not need to enable or disable **LoadPlugins** and **UseD3D8to9** inside global.ini from the scripts folder that comes with the definitive fix. I haven't tested the latest version of the Definitive fix, but I do know they made some big changes to it, though I have no reason to believe they've removed or changed the wrapper.

This is PappaSlask, over and out

![Skärmbild 2024-07-30 152536](https://github.com/user-attachments/assets/dea03166-0014-4128-92bb-3ee11ad263cb)

**Comment by skurtyyskirts** [comment]
You're an absolute legend PappaSlask, that worked! I would've never had thought to try that after trying everything else under the sun haha. It's back up and running! I'm ok with closing the ticket, but want to make sure Nvidia has a chance to jot anything down/take not of anything. 
![Max Payne 2 Screenshot 2024 07 31 - 14 33 48 80](https://github.com/user-attachments/assets/36bcada6-dff1-4c4f-ae5b-7665d05f860d)
